bpdv3/Assets/Scripts/Floor/OnTile.cs

```diff
@@ -2,13 +2,13 @@
 using System.Collections.Generic;
 using UnityEngine;
 
+// This detects when an object has moved on a tile and if it is an enemy.
 public class OnTile : MonoBehaviour
 {
     private void OnTriggerEnter(Collider other)
     {
         if (other.gameObject.CompareTag("enemy") || other.gameObject.CompareTag("NextMove") || other.gameObject.CompareTag("coin"))
```

Comment:
Instead of calling tag of the collider three times you can assign into a vairable before if